### PR TITLE
Add contributing guidelines

### PR DESCRIPTION
**Pull request by pepibumur**
**Issue:** [Link](https://github.com/carambalabs/xcodeproj/issues/75)

Resolves https://github.com/swift-xcode/xcodeproj/issues/75

### Short description
Add contributing guidelines

### Solution
Find a good reference and update it to match xcodeproj guidelines.

### Implementation
- [x] Find a [template](https://github.com/nayafia/contributing-template/blob/master/CONTRIBUTING-template.md) to start from.
- [x] Update it.

### GIF
gif

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Following these guidelines helps to communicate that you respect the time of the developers managing and developing this open source project. In return, they should reciprocate that respect in addressing your issue, assessing changes, and helping you finalize your pull requests.
+
+### A good contribution
+
+xcodeproj is an open source project and we love to receive contributions from our community — you! There are many ways to contribute, from writing tutorials or blog posts, improving the documentation, submitting bug reports and feature requests or writing code which can be incorporated into xcodeproj itself.
+
+### A bad contribution
+
+Please, don't use the issue tracker for support questions. For that purpose use Stack Overflow tagging your threads with `#xcodeproj`.
```
Comment (yonaskolb) [inline CONTRIBUTING.md]: While pushing people to Stack Overflow is good for a large open source project which is fairly stable, for now I think it's worth resolving support here to keep it centralised, and so we're on top of issues.
If that becomes too large then you can point people to Stack Overflow

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+* Keep feature versions as small as possible, preferably one new feature per version.
+* Be welcoming to newcomers and encourage diverse new contributors from all backgrounds.
+* Comply the [code of conduct](https://github.com/swift-xcode/xcodeproj/blob/master/CODE_OF_CONDUCT.md) when you participate in the project.
+
+# Your First Contribution
+
+Unsure where to begin contributing to xcodeproj? You can start by looking through these easy and thread issues:
+
+- **difficulty:easy** - issues which should only require a few lines of code, and a test or two.
+- **type:thread** - issues which should be a bit more involved than beginner issues.
```
Comment (yonaskolb) [inline CONTRIBUTING.md]: I'm not sure what a `thread` type is?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
+
+Working on your first Pull Request? You can learn how from this *free* series, [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+
+At this point, you're ready to make your changes! Feel free to ask for help; everyone is a beginner at first :smile_cat:
+
+# Getting started
+
+1. Create your own fork of the code
```
Comment (yonaskolb) [inline CONTRIBUTING.md]: It's probably worth mentioning that you should run `swift package generate-xcodeproj` with Xcode 8 (will change with Swift 4 update) and then opening the xcode proj

File: CONTRIBUTING.md (hunk 1 above)
Comment (pepibumur) [reply CONTRIBUTING.md]: Good idea, let's keep it in the repo for now. I'll update the document accordingly.

File: CONTRIBUTING.md (hunk 3 above)
Comment (pepibumur) [reply CONTRIBUTING.md]: Good idea 😛, I'll update it.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+xcodeproj is an open source project and we love to receive contributions from our community — you! There are many ways to contribute, from writing tutorials or blog posts, improving the documentation, submitting bug reports and feature requests or writing code which can be incorporated into xcodeproj itself.
+
+### A bad contribution
+
+Please, don't use the issue tracker for support questions. For that purpose use Stack Overflow tagging your threads with `#xcodeproj`.
+
+# Ground Rules
+
+* Ensure that changes in the project are tested.
+* Before merging, all the checks on GitHub should pass.
```
Comment (artemnovichkov) [inline CONTRIBUTING.md]: What did you mean?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+After feedback has been given we expect responses within two weeks. After two weeks we may close the pull request if it isn't showing any activity.
+
+# Community
+
+You can chat with the core team on Slack. You can join the group on the [following link](https://swift-xcode.herokuapp.com).
+
+# Conventions
+
+### Commit message
+- Commit message should include the issue number together with the commit message: `[#number] commit message`.
```
Comment (artemnovichkov) [inline CONTRIBUTING.md]: What if I commit 10 times? Every message should be in this format, right?

Comment (pepibumur) [reply CONTRIBUTING.md]: Yes, it makes it easier to find the issue the commit belonged to, and get some context on that issue.

File: CONTRIBUTING.md (hunk 4 above)
Comment (pepibumur) [reply CONTRIBUTING.md]: Travis-CI mostly. Would you rephrase it differently @artemnovichkov?

**Pull request merged**